File: ndos_create.R
#!/usr/bin/env Rscript
# Create N-DOS project scaffold.
# Usage: Rscript ndos_create.R /path/to/project [--force]

args <- commandArgs(trailingOnly = TRUE)
if (length(args) < 1) {
  stop("Usage: Rscript ndos_create.R /path/to/project [--force]")
}
root <- args[1]
force <- any(args == "--force")

dirs <- c("raw_data","processed_data","analysis","figures","scripts")
if (!dir.exists(root)) dir.create(root, recursive = TRUE, showWarnings = FALSE)
for (d in dirs) {
  p <- file.path(root, d)
  if (dir.exists(p) && !force) next
  dir.create(p, recursive = TRUE, showWarnings = FALSE)
}
readme <- file.path(root, "README.md")
if (!file.exists(readme) || force) {
  writeLines(c("# N-DOS Project", "", "This project follows the N-DOS layout."), readme)
}
cat("[OK] Created/updated N-DOS scaffold at:", root, "\n")

File: ndos_create.py
#!/usr/bin/env python3
Create N-DOS project scaffold.
Usage:
  python ndos_create.py /path/to/project [--force]

import argparse
from pathlib import Path

DEFAULT_DIRS = ["raw_data","processed_data","analysis","figures","scripts"]

def create_project(root: Path, force: bool=False):
    root.mkdir(parents=True, exist_ok=True)
    for d in DEFAULT_DIRS:
        p = root / d
        if p.exists() and not force:
            continue
        p.mkdir(parents=True, exist_ok=True)
    readme = root / "README.md"
    if not readme.exists() or force:
        readme.write_text("# N-DOS Project\n\nThis project follows the N-DOS layout.\n")
    print(f"[OK] Created/updated N-DOS scaffold at: {root}")

def main():
    ap = argparse.ArgumentParser(description="Create N-DOS scaffold")
    ap.add_argument("root", type=Path, help="Project root")
    ap.add_argument("--force", action="store_true", help="Overwrite existing items if present")
    args = ap.parse_args()
    create_project(args.root, args.force)

if __name__ == "__main__":
    main()
